pallets/loans/src/entities/pricing/external.rs

```diff
@@ -150,23 +154,52 @@ impl<T: Config> ExternalActivePricing<T> {
 		}
 	}
 
-	fn linear_accrual_price(&self, maturity: Seconds) -> Result<T::Balance, DispatchError> {
-		Ok(cfg_utils::math::y_coord_in_rect(
-			(self.settlement_price_updated, self.latest_settlement_price),
-			(maturity, self.info.notional),
-			T::Time::now(),
-		)?)
+	fn maybe_with_linear_accrual_price(
+		&self,
+		maturity: Seconds,
+		price: T::Balance,
+		price_last_updated: Seconds,
+	) -> Result<T::Balance, DispatchError> {
+		if self.info.with_linear_pricing {
+			Ok(cfg_utils::math::y_coord_in_rect(
+				(price_last_updated, price),
+				(maturity, self.info.notional),
+				T::Time::now(),
+			)?)
+		} else {
+			Ok(price)
+		}
 	}
 
 	pub fn current_price(
 		&self,
 		pool_id: T::PoolId,
 		maturity: Seconds,
 	) -> Result<T::Balance, DispatchError> {
-		Ok(match T::PriceRegistry::get(&self.info.price_id, &pool_id) {
-			Ok(data) => data.0,
-			Err(_) => self.linear_accrual_price(maturity)?,
-		})
+		self.current_price_inner(
+			maturity,
+			T::PriceRegistry::get(&self.info.price_id, &pool_id).ok(),
+		)
+	}
+
+	fn current_price_inner(
```

Comment:
The current on-chain behavior is:
- If oracle is found, use oracle price
- If not found, use linear accrual price.

That behavior can not be modeled right now. We need to choose between:
- Use oracle price or if not use settlement price (`with_linear_pricing = false`)
- Use always linear accrual price (`with_linear_pricing = true`)

Do we want this? Don't we still want to model the current behavior?

Reply:
cc @mustermeiszer

Reply:
No, we want to change that too. 

Get the price: 
* IF oracle &rarr; use oracle
* ELSE &rarr; use settlement price

Use the price:
* IF linear pricing &rarr; use linear pricing 
* ELSE &rarr; just use price

Reply:
What's the difference between "Get the price" and "Use the price"?

Reply:
Get is either orale or settlement. Use is just for using whatever came out of the get.

Reply:
I would add new 2 test cases to check a loan configured with `use_linear_price = true` always uses the linear accrual with oracle value and with settlement price

Reply:
Well, after the changes of this PR all loans comes with `use_linear_price = true`, so we should check the `false` version instead